FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobDataWriter.cs

```diff
@@ -85,7 +84,10 @@ await foreach (PathItem path in _containerClient.ListPathsAsync(sourceDirectory,
                 if (path.IsDirectory == true)
                 {
                     // Record all directories that need to commit.
-                    Match match = _stagingDataFolderRegex.Match(path.Name);
+                    Match match = _dataSourceType == DataSourceType.FHIR
```

Comment:
use switch to support more data source type in the future.

Reply:
Fixed, thanks!